v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/transforms/ProcessValidateJsonFields.java

```diff
@@ -0,0 +1,208 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.elasticsearch.transforms;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider;
+import com.jayway.jsonpath.spi.mapper.JacksonMappingProvider;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** ProcessValidateJsonFields is used to fix malformed fields, (ex. de-dot). */
+public class ProcessValidateJsonFields
+    extends PTransform<PCollection<String>, PCollection<String>> {
+
+  @Override
+  public PCollection<String> expand(PCollection<String> input) {
+    return input.apply(ParDo.of(new ValidateJsonFieldsFn()));
+  }
+
+  static class ValidateJsonFieldsFn extends DoFn<String, String> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ValidateJsonFieldsFn.class);
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final Configuration configuration =
+        Configuration.builder()
+            .jsonProvider(new JacksonJsonNodeJsonProvider())
+            .mappingProvider(new JacksonMappingProvider())
+            .build();
+
+    /**
+     * "Having a field name with just dot will cause Elasticsearch to throw exception Index -1 out
+     * of bounds for length 0. This is because field like ".": {} is not valid. The error is thrown
+     * on this line because subfields has a length of 0 and hence it is trying to access index -1,
+     * which is not allowed. The method that splits the field names into paths called
+     * splitAndValidatePath and what it does is to split field names when it encounters . (i.e. a
+     * dot). This field is easily found in k8s logs"
+     *
+     * @param context
+     */
+    @ProcessElement
+    public void processElement(ProcessContext context) {
+      String input = context.element();
+      ObjectNode node = null;
+      try {
+        JsonPath.using(configuration)
+            .parse(input)
+            .read(
+                "$.protoPayload.response.spec.template.spec.containers.livenessProbe.httpGet.port");
+      } catch (Exception ignored) {
+      }
+
+      JsonNode requestApp = null;
```

Comment:
Since there is a lot of repeated code, add functions shown below and call the functions for different jpaths (i.e requestAPP, responseApp etc)
1. parseJson(String json, String jpath) to handle read functionality
2. updateJson(String json, ... other parameters) to handle put functionality
3. deleteJsonString json, String jpath) to handle delete functionality